src/docs/development/platform-integration/platform-views.md

```diff
@@ -196,58 +243,299 @@ public class MainActivity extends FlutterActivity {
+}
+
+@end
+```
+
+Finally, register the platform view. This can be done in an app or a plugin.
+
+
+For app registration, modify the App's `AppDelegate.m`:
+
+```objc
+#import "AppDelegate.h"
+#import "FLNativeView.h"
+#import "GeneratedPluginRegistrant.h"
+
+@implementation AppDelegate
+
+- (BOOL)application:(UIApplication *)application
+    didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
+  [GeneratedPluginRegistrant registerWithRegistry:self];
+
+   NSObject<FlutterPluginRegistrar>* registrar =
+      [self registrarForPlugin:@"plugin-name"];
+
+  FLNativeViewFactory* factory =
+      [[FLNativeViewFactory alloc] initWithMessenger:registrar.messenger];
+  [[self registrarForPlugin:@"<plugin-name>"] registerViewFactory:factory
+                                                          withId:@"<platform-view-type>"];
+  return [super application:application didFinishLaunchingWithOptions:launchOptions];
+}
+
+@end
+```
+
+For plugin registration, modify the main plugin file (e.g. `FLPlugin.m`):
+
+```objc
+#import "FLPlugin.h"
+#import "FLNativeView.h"
+
+@implementation FLPlugin
+
++ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
+  FLNativeViewFactory* factory =
+      [[FLNativeViewFactory alloc] initWithMessenger:registrar.messenger];
+  [registrar registerViewFactory:factory withId:@"<platform-view-type>"];
+}
+
+@end
+```
+
+For more information, see the API docs for:
+
+* [`FlutterPlatformViewFactory`][]
+* [`FlutterPlatformView`][]
+* [`PlatformView`][]
+
+[`FlutterPlatformViewFactory`]: {{site.api}}/objcdoc/Protocols/FlutterPlatformViewFactory.html
+[`FlutterPlatformView`]: {{site.api}}/objcdoc/Protocols/FlutterPlatformView.html
+[`FlutterPlugin`]: {{site.api}}/objcdoc/Protocols/FlutterPlugin.html
 [`UIKitView`]: {{site.api}}/flutter/widgets/UiKitView-class.html
+
+## Putting it together
+
+When implementing the `build()` method in Dart, you can use [`defaultTargetPlatform`][]
+to detect the platform, and decide what widget to use:
+
+<!-- skip -->
+```dart
+Widget build(BuildContext context) {
+  // This is used in the platform side to register the view.
+  final String viewType = '<platform-view-type>';
+  // Pass parameters to the platform side.
+  final Map<String, dynamic> creationParams = <String, dynamic>{};
+
+  switch (defaultTargetPlatform) {
+    case TargetPlatform.android:
+      // return widget on Android.
+    case TargetPlatform.iOS:
+      // return widget on iOS.
+    default:
+      throw UnsupportedError("Unsupported platform view");
+  }
+}
+```
+
+[`defaultTargetPlatform`]: {{site.api}}/flutter/foundation/defaultTargetPlatform.html
+
+
+### Performance
+
+Platform views in Flutter come with performance trade-offs.
+
+For example, in a typical Flutter app, the Flutter UI is composed
+on a dedicated raster thread. This allows Flutter apps to be fast,
+as the main platform thread is rarely blocked.
+
+While a platform view is rendered, the Flutter UI is composed from
```

Comment:
Is this always the case or only with hybrid composition?

Reply:
Done.